### PR TITLE
safekeeper: fsync filesystem on start

### PR DESCRIPTION
**Pull request by arssher**

### CODE REVIEW

File: safekeeper/src/bin/safekeeper.rs [hunk 1]
```diff
@@ -261,6 +261,15 @@ async fn main() -> anyhow::Result<()> {
...
 
+    // Prevent running multiple safekeepers on the same directory
+    let lock_file_path = workdir.join(PID_FILE_NAME);
+    let lock_file =
+        pid_file::claim_for_current_process(&lock_file_path).context("claim pid file")?;
+    info!("claimed pid file at {lock_file_path:?}");
+    // ensure that the lock file is held even if the main thread of the process is panics
+    // we need to release the lock file only when the current process is gone
+    std::mem::forget(lock_file);
+
```
Comment (problame) [inline safekeeper/src/bin/safekeeper.rs:272]: This seems unrelated

**Review by problame — APPROVED** [review]

**Review by arpad-m — APPROVED** [review]

File: safekeeper/src/bin/safekeeper.rs (hunk 1 above)
Comment (arssher) [reply safekeeper/src/bin/safekeeper.rs:272]: Did it in separate commit, it is small and in related place.
